File: PRM/Piwne_Przepisy/app/src/main/java/com/s17918/piwneprzepisy/MainActivity.java
package com.s17918.piwneprzepisy;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Spinner;
import android.widget.TextView;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void OpenRecipe(View view){

        Spinner beer_spinner = (Spinner) findViewById(R.id.beer_name_spinner);
        String beer_name = String.valueOf(beer_spinner.getSelectedItem());

        long id = beer_spinner.getSelectedItemId();

        GetVariables.getBeerName_Brew(beer_name);
        GetVariables.getBeerId_Brew(id);

        TextView SpinnerError = (TextView)findViewById(R.id.SpinnerError);

        if(id!=0){
            Intent intent = new Intent(this, IngredientsActivity.class);
            startActivity(intent);
            SpinnerError.setText(R.string.noerror);
        }else {
            SpinnerError.setText(R.string.error);
        }


    }
}

File: Kotlin/RollDice/app/src/main/java/pjatk/prm/s17918/rolldice/MainActivity.kt
package pjatk.prm.s17918.rolldice

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button: Button = findViewById(R.id.button)
        button.setOnClickListener{
            rollDice()
        }

    }

    private fun rollDice(){
        val dice = Dice(6)
        val diceRoll = dice.roll()
        val text: TextView = findViewById(R.id.text)
        text.text = diceRoll.toString()
    }
}

class Dice(private val sides: Int){
    fun roll(): Int{return (1..sides).random()}
}
